daemons/controld/controld_execd.c

```diff
@@ -1648,6 +1648,18 @@ construct_op(const lrm_state_t *lrm_state, const xmlNode *rsc_op,
         op->interval_ms = 0;
     }
 
+    /* If this is a start action with the special expire_attrs meta-attribute
+     * set to false, go ahead and clear the purge bit from from the connection
+     * resource's flags.
+     */
+    if (!crm_is_true(crm_meta_value(params, PCMK_META_ATTR_EXPIRE))) {
```

Comment:
crm_is_true() defaults to false if the string is NULL -- we want to consider it true for this purpose. Maybe it would be easier to make it preserve_attributes=true